submissions/nadsatt/friends-app/modules/page-link-list.js

```diff
@@ -0,0 +1,151 @@
+import { PageLink, CornerPageLink } from './page-link-items.js';
+
+export class PageLinkList {
+    constructor(userService, userCardList){
+        this.userService = userService;
+        this.userCardList = userCardList;
+        this.fragment = new DocumentFragment();
+
+        this.usersPerPage = 10;
+        this.maxPageLinksNumber = 5;
+
+        this.defineCornerPageLinks();
+        this.defineElement();
+        this.insertElement();
+        this.defineElementClickHandler();
+    }
+
+    defineCornerPageLinks(){
+        this.leftPageLinks = ['First', 'Prev'].map(value => new CornerPageLink(value).element);
+        this.rightPageLinks = ['Next', 'Last'].map(value => new CornerPageLink(value).element);
+    }
+
+    defineElement(){
+        this.element = document.createElement('ul');
+        this.element.classList.add('page-link-list');
+    }
+
+    insertElement(){
+        document.querySelector('.page-link-list-wrapper').append(this.element);
+    }
+
+    defineElementClickHandler(){
+        this.element.onclick = ({target, target:{textContent: value}}) => {
+            if(this.checkIfMiddlePageLinkClicked(target)){
+                let currentPageNumber = +value;
+                this.performPagination(currentPageNumber);
+            }
+            else if(this.checkIfCornerPageLinkClicked(target)){
+                let currentPageNumber = value === 'First' ? 1 :
+                                        value === 'Prev' ? --this.currentPageNumber :
+                                        value === 'Next' ? ++this.currentPageNumber :
+                                        this.pagesNumber;
+                this.performPagination(currentPageNumber);
+            }
+        }
+    }
+
+    checkIfMiddlePageLinkClicked(target){
+        return target.classList.contains('page-link-item');
+    }
+
+    checkIfCornerPageLinkClicked(target){
+        return target.classList.contains('corner-page-link-item');
+    }
+
+    performPagination(currentPageNumber = 1){
+        this.getUsersNumber();
+        this.definePagesNumber();
+
+        this.validateCurrentPageNumber(currentPageNumber);
+        this.defineCurrentPageUsers();
+        this.updateCurrentPageUsers();
+        this.userCardList.updateUserCards();
+
+        this.defineFirstAndLastMiddlePageLinksNumbers();
+        this.defineMiddlePageLinksNumbers();
+
+        this.defineMiddlePageLinks();
+        this.updateCornerPageLinks();
+        this.renderPageLinks();
+    }
+
+    getUsersNumber(){
+        this.usersNumber = this.userService.users.length;
```

Comment:
This number can be changed? You will re-get this data on every call of pagination

Reply:
@zonzujiro `users` variable contains filtered (or not - if no filters were applied) users

Maybe i should rename it to make it more evident

Reply:
`usersOnPage` maybe?